diff --git a/src/pcbnew_do b/src/pcbnew_do
index d35e5d2..618b853 100755
--- a/src/pcbnew_do
+++ b/src/pcbnew_do
@@ -1206,7 +1206,7 @@ if __name__ == '__main__':
                            cfg.input_file]
                 else:
                     cmd = [cfg.pcbnew, cfg.input_file]
-                with PopenContext(cmd, stderr=flog_err, close_fds=True,
+                with PopenContext(["vglrun", "-d", ":1"] + cmd, stderr=flog_err, close_fds=True,
                                   stdout=flog_out, start_new_session=True) as pcbnew_proc:
                     cfg.pcbnew_pid = pcbnew_proc.pid
                     cfg.popen_obj = pcbnew_proc
